### PR TITLE
[WIP] Add pytorch variants

### PR DESCRIPTION
**Pull request by k-dominik**
- added conda_build_config.yaml in order to set up "matrix-build"
- modified meta.yaml
  - to use variables from conda_build_config, and
  - to build two separate packages, inferno, infernocpu with pytorch/torchvision
    and pytorch-cpu/torchvision-cpu as depdendencies. This is done by defining three lists, `package_name`, `pytorch_flavor`, and `torchvision_flavor` and zipping them via the special `zip_keys`.

There are three reasons why I marked this thing [WIP]:

1) I'm not sure how to get around the initialization of variables in the beginning of the script, but it is necessary because `conda-build` seems to render the recipe first without taking onto account `conda_build_config.yaml`
2) The pinning for `pytorch` and `torchvision` is now done in a very explicit way with specified `min` and `max` version in `conda_build_config. In order to use `pin_run_as_build` one would have to add `pytorch` and `torchvision` to the `host` environment and they would, thus, be downloaded at build time, for no real reason other to use the more elegant pinning. So I'm happy to hear some opinions here.
3) `conda-smithy rerender` [won't add custom channels atm](https://github.com/conda-forge/conda-smithy/issues/935).

Checklist
* [x] Used a fork of the feedstock to propose changes
* [x] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/conda_smithy.html#how-to-re-render ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by k-dominik** [comment]
@conda-forge-admin, please rerender

**Comment by k-dominik** [comment]
@conda-forge-admin, please rerender

**Comment by constantinpape** [comment]
@k-dominik 
FYI, I updated the recipe so this is build as `noarch: python`.
Don't know how this interacts with this PR, but would be nice to have the `inferno-cpu` package :).

**Comment by k-dominik** [comment]
Okay, I still have problems getting the custom channels play with conda-smithy. It's find to build the two recipes with

```
conda-build -c pytorch -c conda-forge recipe
```

I'll upload these packages to `ilastik-forge` for now.

**Comment by k-dominik** [comment]
okay, closing this, `pytorch` flavor should be explicitly given when installing...

e.g. `conda install -c pytorch -c conda-forge pytorch-cpu inferno`

_update: okay, this, of course, will then, install both, `pytorch` and `pytorch-cpu`, which is kind of unnecessary. Reopening this and waiting until the channel thing is resolved_

File: recipe/conda_build_config.yaml [hunk 1]
```diff
@@ -0,0 +1,20 @@
+channel_sources:
+  - pytorch,conda-forge,defaults
```
Comment (isuruf) [inline recipe/conda_build_config.yaml:2]: Please don't use `pytorch` channel here. conda-forge doesn't support additional channels other than defaults. (even though conda-smithy supports them, conda-forge doesn't.)

Comment (k-dominik) [reply recipe/conda_build_config.yaml:2]: Oh thank you very much for the heads-up on this @isuruf . I'll close this PR then and build/upload on our own channel then.

**Comment by k-dominik** [comment]
Closed, since `conda-forge` doesn't allow for other channels as sources.

**Pull request closed without merging**